types/jest/index.d.ts

```diff
@@ -393,8 +393,11 @@ declare namespace jest {
     function spyOn<T extends {}, M extends ConstructorPropertyNames<Required<T>>>(
         object: T,
         method: M,
-    ): Required<T>[M] extends new (...args: any[]) => any
-        ? SpyInstance<InstanceType<Required<T>[M]>, ConstructorArgsType<Required<T>[M]>>
+    ): ConstructorProperties<Required<T>>[M] extends new (...args: any[]) => any
```

Comment:
basically the idea here is to use the same approach as with functions, but also remove index signatures because we can be pretty much sure that they are not constructors I think